### PR TITLE
Switch from (crypto-secure) rand to fastrand

### PR DESCRIPTION
**Pull request by red15**
fastrand should be a lot better suited, less hassle with threads trying to optimize random selections.
did not test the threaded build tbh but should be ok.

### CODE REVIEW

**Comment by avinassh** [comment]
@red15 could you test, post the before and after numbers?

**Comment by red15** [comment]
> @red15 could you test, post the before and after numbers?

I can't I just cobbled this together on a windows pc and don't have the proper tools to do the timing comparison, I did build and verify with a rust compiler that the code is valid.

**Comment by avinassh** [comment]
no problem at all @red15 :) 

Let me benchmark it, post the numbers here.

**Comment by red15** [comment]
Ok got around to running this as intended:

`make busy-rust`

```
Master b6671ce94
real    0m11.400s | real    0m11.189s | real    0m10.717s
user    0m10.430s | user    0m10.113s | user    0m9.651s
sys      0m0.963s | sys     0m1.069s  | sys     0m1.060s
```

```
fastrand
real    0m9.683s | real    0m9.836s | real    0m9.334s
user    0m8.607s | user    0m8.835s | user    0m8.270s
sys     0m1.065s | sys     0m0.993s | sys     0m1.056s
```

`make busy-rust-thread`

```
Master b6671ce94
real  0m3.707s | real  0m3.966s | real  0m3.473s
user 0m11.510s | user 0m13.007s | user 0m11.566s
sys   0m1.220s | sys   0m1.111s | sys   0m1.167s
```

```
fastrand
real  0m3.459s | real 0m3.531s | real 0m2.866s
user 0m10.758s | user 0m9.882s | user 0m9.517s
sys   0m1.139s | sys  0m1.124s | sys  0m1.175s
```
Not a super big difference but noticeable anyway I'd say ?

File: src/bin/common.rs [hunk 1]
```diff
@@ -1,23 +1,18 @@
...
 
 pub fn get_random_bool() -> bool {
-    rand::random()
+    fastrand::bool()
 }
 
 pub fn get_random_area_code() -> String {
-    let mut rng = rand::thread_rng();
-    format!("{:06}", rng.gen_range(0..999999))
+    format!("{:06}", fastrand::u32(0..999_999))
```
Comment (red15) [inline src/bin/common.rs:17]: Using `perf record ./target/release/busy` I found another optimalization that can be done, the `Display::Fmt` trait is being called quite often for the area code generation, changing it basically to just become 

```rust
    fastrand::u32(100_000..999_999).to_string()
```

saves us from the rather slow `Display::Fmt`, I can include that change too if you like,

Some numbers of fastrand + display_fmt fix in `make busy-rust`
```
real 0m8.211s
user 0m7.208s
sys  0m0.996s
```

and `make busy-rust-thread`
```
real 0m2.605s
user 0m8.691s
sys  0m1.211s
```

**Comment by avinassh** [comment]
Any reason you just ran for busy threads, but not for actual insertions?

**Comment by avinassh** [comment]
I ran this on actual insertions, this reduced 2s further. Thank you!

**Comment by avinassh** [comment]
```sh
Sun Aug  8 13:52:32 IST 2021 [RUST] basic_batched.rs (100_000_000) inserts

real	0m32.424s
user	0m30.826s
sys	0m2.272s

Sun Aug  8 13:53:06 IST 2021 [RUST] threaded_batched.rs (100_000_000) inserts

real	0m30.094s
user	0m42.704s
sys	0m3.877s
```

**Review by red15 — COMMENTED** [review]
> Any reason you just ran for busy threads, but not for actual insertions?

Not sure how to do that? Is that in the Makefile somewhere ?

> I ran this on actual insertions, this reduced 2s further. Thank you!

Good to see it matches the improvements I was seeing.

**Pull request merged**